### PR TITLE
add mev-boost

### PR DESCRIPTION
**Pull request by xenowits**
Adds `mev-boost` container.

category: feature
ticket: https://github.com/ObolNetwork/charon-distributed-validator-node/issues/158

### CODE REVIEW

File: mevboost-compose.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  #  | | | | | |  __/\ V /_____| |_) | (_) | (_) \__ \ |_
+  #  |_| |_| |_|\___| \_/      |_.__/ \___/ \___/|___/\__|
+  mev-boost:
+    image: flashbots/mev-boost:${MEVBOOST_VERSION:-v1.4.0}
+    command: |
+      -goerli 
+      -loglevel debug
+      -addr 0.0.0.0:18550
+      -relay-check
+      -relays "https://0x8f7b17a74569b7a57e9bdafd2e159380759f5dc3ccbd4bf600414147e8c4e1dc6ebada83c0139ac15850eb6c975e82d0@builder-relay-goerli.blocknative.com,https://0xafa4c6985aa049fb79dd37010438cfebeb0f2bd42b115b89dd678dab0670c1de38da0c4e9138c9290a398ecd9a0b3110@boost-relay-goerli.flashbots.net"
```
Comment (xenowits) [inline mevboost-compose.yml:21]: Find public relays here: https://research.lido.fi/t/lido-on-ethereum-call-for-relay-providers/2844

File: docker-compose.yml [hunk 2]
```diff
@@ -56,6 +56,7 @@ services:
       --execution-endpoint=http://geth:8551
       --execution-jwt=/opt/jwt/jwt.hex
       --datadir=/opt/app/beacon/
+      --builder=http://mev-boost:18550
```
Comment (xenowits) [inline docker-compose.yml:59]: https://lighthouse-book.sigmaprime.io/builders.html#how-to-connect-to-a-builder

File: mevboost-compose.yml (hunk 1 above)
Comment (corverroos) [reply mevboost-compose.yml:22]: add a comment where you got this

File: docker-compose.yml (hunk 2 above)
Comment (corverroos) [reply docker-compose.yml:59]: can add this be default since mevboost probably not running

File: mevboost-compose.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    ports:
+      - ${MEVBOOST_HTTP_PORT:-18550}:18550
```
Comment (corverroos) [inline mevboost-compose.yml:24]: no need to bind mount by default

File: mevboost-compose.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    networks: [ dvnode ]
```
Comment (corverroos) [inline mevboost-compose.yml:25]: this should be a variable

File: mevboost-compose.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+# Override any defaults specified by `${FOO:-bar}` in `.env` with `FOO=qux`.
+
+services:
+  #      _                     _
+  #  _ __ ___   _____   __    | |__   ___   ___  ___| |_
+  #  | '_ ` _ \ / _ \ \ / /____| '_ \ / _ \ / _ \/ __| __|
+  #  | | | | | |  __/\ V /_____| |_) | (_) | (_) \__ \ |_
+  #  |_| |_| |_|\___| \_/      |_.__/ \___/ \___/|___/\__|
+  mev-boost:
```
Comment (corverroos) [inline mevboost-compose.yml:15]: need docs for how to opt-into this

File: mevboost-compose.yml (hunk 1 above)
Comment (xenowits) [reply mevboost-compose.yml:22]: this is already added on line 5

File: mevboost-compose.yml (hunk 3 above)
Comment (xenowits) [reply mevboost-compose.yml:24]: Removed default port binding. Also added port binding to override file

File: docker-compose.yml (hunk 2 above)
Comment (xenowits) [reply docker-compose.yml:59]: When mevboost is not running, lighthouse doesn't actually error

Comment (corverroos) [reply docker-compose.yml:59]: as ok, fine then

File: docker-compose.override.yml.sample [hunk 6]
```diff
@@ -53,3 +53,7 @@
     # Bind prometheus internal ports to host ports
     #ports:
       #- 9090:9090 # Metrics
+
+  #mev-boost:
+    #ports:
```
Comment (corverroos) [inline docker-compose.override.yml.sample:58]: add comment: "    # Bind prometheus internal ports to host ports"

**Review by corverroos — APPROVED** [review]

File: mevboost-compose.yml (hunk 5 above)
Comment (xenowits) [reply mevboost-compose.yml:15]: The docs PR is also merged now

**Pull request merged**